deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkUri.java

```diff
@@ -386,7 +390,48 @@ Builder newBuilder() {
   public static DeepLinkUri parse(String url) {
     Builder builder = new Builder();
     Builder.ParseResult result = builder.parse(null, url);
-    return result == Builder.ParseResult.SUCCESS ? builder.build() : null;
+    if (result == Builder.ParseResult.SUCCESS) {
+      lastBuilder = builder;
+      return builder.build();
+    } else {
+      lastBuilder = null;
+      return null;
+    }
+  }
+
+  /**
+   * Replace path segments that match pathVariableReplacementValue with the pathVariablePlaceholder
+   * that would be present in a DeepLinkEntry. {@linkplain #PATH_VARIABLE_PLACEHOLDER_VALUE}.
+   * <p/>
+   * Given:
+   * <ul>
+   *   <li>path segments like: "/obamaOs/cereals"</li>
+   *   <li>pathVariableReplacementValue = "obamaOs"</li>
+   * </ul>
+   * Then produce URI with path segments like:
+   * <li><xmp>/-replaceable-path-variable-/users/{param1}")</xmp></li>
+   * @return URI or null if no replacements could be made.
+   */
+  public static DeepLinkUri parseWithReplacement(String pathVariableReplacementValue) {
+    if (lastBuilder == null
+      || pathVariableReplacementValue == null
+      || pathVariableReplacementValue.isEmpty()
+    ) {
+      return null;
+    }
+    List<String> segments = lastBuilder.encodedPathSegments;
```

Comment:
So we are expecting that this is in a certain state (and not the instance but the class)?